### ISSUE TITLE
[trainer] Simplify Trainer dependencies by making TrainerTrainingTricksMixin a utils class

### ISSUE THREAD

**Issue by ananthsub** [PyTorchLightning/pytorch-lightning#6815]
## 🚀 Feature

Training tricks doesn't need to be inherited from the core Trainer class. These methods could be utility functions that sit completely outside the Trainer class hierarchy. 
https://github.com/PyTorchLightning/pytorch-lightning/blob/bb9ace43334ad50e3758d9cff08ad34216c7d4da/pytorch_lightning/trainer/training_tricks.py#L28

### Motivation

Simplify the Trainer inheritance hierarchy and create more reusable utilities

### Pitch

Deprecate the mixin and move the functions out to a utilities file for NaN detection. All we have to do is accept an nn.Module as input and pass the LightningModule to these functions from within the training loop.

1.
```
 def print_nan_gradients(model: nn.Module) -> None:
```
2.
```
def detect_nan_tensors(model: nn.Module) -> None:
```
we can separately check for the loss being NaN inside the training loop directly, since thats a specific call here: https://github.com/PyTorchLightning/pytorch-lightning/blob/bb9ace43334ad50e3758d9cff08ad34216c7d4da/pytorch_lightning/trainer/training_tricks.py#L43-L45

Deprecation path:
- Add the new utilities in a standalone file
- Make the mixin call the new utilities
- Update the training loop to directly call the new utilities
- Add a deprecation warning to the methods inside the mixin that mark them as deprecated

### Alternatives

Keep as is

**Comment by ananthsub** [comment]
same goes for https://github.com/PyTorchLightning/pytorch-lightning/blob/22a266d8b8cf57455cc863e20491e416ec635ba7/pytorch_lightning/trainer/logging.py#L22
